sqlmesh/lsp/main.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python
+"""A Language Server Protocol (LSP) server for SQL with SQLMesh integration."""
+
+import logging
+import typing as t
+from contextlib import suppress
+from pathlib import Path
+
+from sqlmesh.core.context import Context
+from lsprotocol import types
+from pygls.server import LanguageServer
+from pygls.workspace import TextDocument
+from sqlmesh._version import __version__
+from sqlmesh.core.dialect import Audit
+from sqlmesh.core.model import Model
+
+logger = logging.getLogger(__name__)
+
+GLOBAL_CONTEXT: t.Optional[
```

Comment:
why is this a tuple, would it be easier for this to be two separate variables?

Reply:
Changed, it's mainly where I moved from, I kind of like the aspect of not having a empty dict before you can have Context.